trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/ParsedRequest.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server.rest;
+
+import jakarta.ws.rs.core.MultivaluedMap;
+
+import java.net.URI;
+
+import static java.util.Objects.requireNonNull;
+
+public record ParsedRequest(URI requestUri, MultivaluedMap<String, String> requestHeaders, MultivaluedMap<String, String> requestQueryParameters, String httpVerb)
+{
+    public ParsedRequest
+    {
+        requireNonNull(requestUri, "requestUri is null");
+        requireNonNull(httpVerb, "httpVerb is null");
```

Comment:
Nit: I believe we usually keep the order of checking of items in the constructor the same as in the signature. So should `httpVerb` be checked last?